src/coreclr/jit/emitarm64.cpp

```diff
@@ -11731,31 +11872,87 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
                 vt              = tmpDsc->tdTempType();
             }
             if (vt == TYP_REF || vt == TYP_BYREF)
+            {
                 emitGCvarDeadUpd(adr + ofs, dst DEBUG_ARG(varNum));
+            }
         }
         if (emitInsWritesToLclVarStackLocPair(id))
         {
-            unsigned ofs2 = ofs + TARGET_POINTER_SIZE;
+            int      varNum2 = varNum;
+            int      adr2    = adr;
+            unsigned ofs2    = ofs;
+            unsigned ofs2Dist;
+
+            if (id->idIsLclVarPair())
+            {
+                bool FPbased2;
+                if (id->idSmallCns())
+                {
+                    varNum2 = ((instrDescLclVarPair*)id)->iiaLclVar2.lvaVarNum();
+                    ofs2    = ((instrDescLclVarPair*)id)->iiaLclVar2.lvaOffset();
+                }
+                else
+                {
+                    varNum2 = ((instrDescLclVarPairCns*)id)->iiaLclVar2.lvaVarNum();
+                    ofs2    = ((instrDescLclVarPairCns*)id)->iiaLclVar2.lvaOffset();
+                }
+
+                // If there are 2 GC vars in this instrDesc, get the 2nd variable
+                // that should be tracked.
+                adr2     = emitComp->lvaFrameAddress(varNum2, &FPbased2);
+                ofs2Dist = EA_SIZE_IN_BYTES(size);
+#ifdef DEBUG
+                assert(FPbased == FPbased2);
+                if (FPbased)
+                {
+                    assert(id->idReg3() == REG_FP);
+                }
+                else
+                {
+                    assert(id->idReg3() == REG_SP);
+                }
+                assert(varNum2 != -1);
+#endif // DEBUG
+            }
+            else
+            {
+                ofs2Dist = TARGET_POINTER_SIZE;
+                ofs2 += ofs2Dist;
+            }
+
+            ofs2 = AlignDown(ofs2, ofs2Dist);
```

Comment:
I still don't understand why the `AlignDown` is necessary.

Reply:
thinking again, don't seem to be needed, but didn't you suggested it in https://github.com/dotnet/runtime/pull/85032#discussion_r1173991069?

Reply:
Well, I guess I added it in that example code to be consistent with the code immediately prior that uses it. When we were chatting about this PR once I think I mentioned that I didn't understand the need for AlignDown. If nothing else, if you figure out why we need it, it would be good to comment it.

Reply:
I will send a follow up PR to add the comment.